File: SuperiorPackGroup/shipmentConditionXtraForm.vb
Public Class shipmentConditionXtraForm 

    Private Sub shipmentConditionXtraForm_Load(sender As Object, e As EventArgs) Handles Me.Load

        checkpermissions()

    End Sub

    Private Sub deleteSimpleButton_Click(sender As Object, e As EventArgs) Handles deleteSimpleButton.Click

        If conditionsGridView.SelectedRowsCount > 0 Then
            If MessageBox.Show("Are you sure you want to delete this record/s?", "Delete Record", MessageBoxButtons.YesNo, MessageBoxIcon.Question) = Windows.Forms.DialogResult.Yes Then
                For i As Integer = 0 To conditionsGridView.SelectedRowsCount - 1
                    Dim row As Integer = conditionsGridView.GetSelectedRows()(i)
                    If row >= 0 Then
                        If ShipmentConditionsBLL.DeleteCondition(CType(conditionsGridView.GetRowCellValue(row, colOid), Integer)) <> True Then
                            MessageBox.Show("The Condition/s was not deleted succesfully.", "Error Encountered", MessageBoxButtons.OK, MessageBoxIcon.Error)
                            Exit Sub
                        End If
                    End If
                Next i
            End If
        End If

        shipmentConditionsXpCollection.Reload()
        conditionsGridView.RefreshData()

    End Sub

    Public Function CheckPermissions() As Boolean

        Try
            Dim permission As SPG.UserPermissionsRow = New UserPermissionsBLL().GetUserPermissionsByUserName(My.Settings.UserName, "General Lists")(0)

            Select Case permission.PermissionLevel
                Case "FULL", "Full"
                    conditionsGridView.OptionsBehavior.Editable = True
                    conditionsGridView.OptionsView.NewItemRowPosition = DevExpress.XtraGrid.Views.Grid.NewItemRowPosition.Bottom
                    Me.deleteSimpleButton.Visible = True
                Case "ADD", "Add"
                    conditionsGridView.OptionsBehavior.Editable = True
                    conditionsGridView.OptionsView.NewItemRowPosition = DevExpress.XtraGrid.Views.Grid.NewItemRowPosition.Bottom
                    Me.deleteSimpleButton.Visible = False
                Case "NONE", "None", "VIEW ASSIGNED", "EDIT ASSIGNED"
                    MessageBox.Show("You don't have permission to view this form.", "User Permissions", MessageBoxButtons.OK, MessageBoxIcon.Hand)
                    Me.Close()
                    Return False
                Case "VIEW", "View"
                    conditionsGridView.OptionsBehavior.Editable = False
                    conditionsGridView.OptionsView.NewItemRowPosition = DevExpress.XtraGrid.Views.Grid.NewItemRowPosition.None
                    Me.deleteSimpleButton.Visible = False
                Case "EDIT", "Edit"
                    conditionsGridView.OptionsBehavior.Editable = True
                    conditionsGridView.OptionsView.NewItemRowPosition = DevExpress.XtraGrid.Views.Grid.NewItemRowPosition.None
                    Me.deleteSimpleButton.Visible = False
                Case Else
                    MessageBox.Show("You don't have permission to view this form.", "User Permissions", MessageBoxButtons.OK, MessageBoxIcon.Hand)
                    Me.Close()
                    Return False
            End Select
            Return True
        Catch ex As InvalidOperationException
            Return False
        End Try

    End Function

End Class

File: SuperiorPackGroup/shipmentConditionXtraForm.cs
//INSTANT C# NOTE: Formerly VB project-level imports:
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Drawing;
using System.Diagnostics;
using System.Windows.Forms;

namespace SuperiorPackGroup
{
	public partial class shipmentConditionXtraForm
	{

		public shipmentConditionXtraForm()
		{
			InitializeComponent();
		}

		private void shipmentConditionXtraForm_Load(object sender, EventArgs e)
		{

			CheckPermissions();

		}

		private void deleteSimpleButton_Click(object sender, EventArgs e)
		{

			if (conditionsGridView.SelectedRowsCount > 0)
			{
				if (MessageBox.Show("Are you sure you want to delete this record/s?", "Delete Record", MessageBoxButtons.YesNo, MessageBoxIcon.Question) == DialogResult.Yes)
				{
					for (int i = 0; i < conditionsGridView.SelectedRowsCount; i++)
					{
						int row = conditionsGridView.GetSelectedRows()[i];
						if (row >= 0)
						{
							if (ShipmentConditionsBLL.DeleteCondition(Convert.ToInt32(conditionsGridView.GetRowCellValue(row, colOid))) != true)
							{
								MessageBox.Show("The Condition/s was not deleted succesfully.", "Error Encountered", MessageBoxButtons.OK, MessageBoxIcon.Error);
								return;
							}
						}
					}
				}
			}

			shipmentConditionsXpCollection.Reload();
			conditionsGridView.RefreshData();

		}

		public bool CheckPermissions()
		{

			try
			{
				SPG.UserPermissionsRow permission = (new UserPermissionsBLL()).GetUserPermissionsByUserName(Properties.Settings.Default.UserName, "General Lists")[0];

				switch (permission.PermissionLevel)
				{
					case "FULL":
					case "Full":
						conditionsGridView.OptionsBehavior.Editable = true;
						conditionsGridView.OptionsView.NewItemRowPosition = DevExpress.XtraGrid.Views.Grid.NewItemRowPosition.Bottom;
						deleteSimpleButton.Visible = true;
						break;
					case "ADD":
					case "Add":
						conditionsGridView.OptionsBehavior.Editable = true;
						conditionsGridView.OptionsView.NewItemRowPosition = DevExpress.XtraGrid.Views.Grid.NewItemRowPosition.Bottom;
						deleteSimpleButton.Visible = false;
						break;
					case "NONE":
					case "None":
					case "VIEW ASSIGNED":
					case "EDIT ASSIGNED":
						MessageBox.Show("You don't have permission to view this form.", "User Permissions", MessageBoxButtons.OK, MessageBoxIcon.Hand);
						Close();
						return false;
					case "VIEW":
					case "View":
						conditionsGridView.OptionsBehavior.Editable = false;
						conditionsGridView.OptionsView.NewItemRowPosition = DevExpress.XtraGrid.Views.Grid.NewItemRowPosition.None;
						deleteSimpleButton.Visible = false;
						break;
					case "EDIT":
					case "Edit":
						conditionsGridView.OptionsBehavior.Editable = true;
						conditionsGridView.OptionsView.NewItemRowPosition = DevExpress.XtraGrid.Views.Grid.NewItemRowPosition.None;
						deleteSimpleButton.Visible = false;
						break;
					default:
						MessageBox.Show("You don't have permission to view this form.", "User Permissions", MessageBoxButtons.OK, MessageBoxIcon.Hand);
						Close();
						return false;
				}
				return true;
			}
			catch (InvalidOperationException ex)
			{
				return false;
			}

		}

	}
}